packages/api/src/hooks/useAllWalletAccounts.ts

```diff
@@ -0,0 +1,72 @@
+import { useMemo } from 'react';
+import useCurrencyConfig from './useCurrencyConfig';
+import useWalletAccountsList from './useWalletAccountsList';
+import useAvailableWallets from './useAvailableWallets';
+
+/** A custom hook that gets the list of added and non-added wallet accounts. */
+const useAllWalletAccounts = () => {
+    const { data: non_added_wallets, ...rest } = useAvailableWallets();
+    const { data: added_wallets } = useWalletAccountsList();
+    const { getConfig } = useCurrencyConfig();
+
+    /** Get all the added and non-added wallets account */
+    const modified_wallets = useMemo(() => {
+        const non_virtual_wallets = added_wallets?.filter(wallet => !wallet.is_virtual);
+
+        /** Modify added wallets list with adding `is_added` and `is_crypto` flag */
+        const modified_added_wallets = non_virtual_wallets?.map(wallet => ({
+            currency: wallet.currency,
+            landing_company_name: wallet.landing_company_name,
+            is_added: true,
+            is_crypto: wallet.currency ? getConfig(wallet.currency)?.is_crypto : false,
+        }));
+
+        /** Modify non-added wallets list with adding `is_added` and `is_crypto` flag */
+        const modified_non_added_wallets = non_added_wallets?.map(wallet => ({
+            currency: wallet.currency,
+            landing_company_name: wallet.landing_company,
+            is_added: false,
+            is_crypto: wallet.account_type === 'crypto',
+        }));
+
+        return [...(modified_non_added_wallets || []), ...(modified_added_wallets || [])];
+    }, [non_added_wallets, added_wallets, getConfig]);
+
+    /** Sort the wallets by fiat, crypto, then virtual */
+    const sorted_wallets = useMemo(() => {
+        if (!modified_wallets) return;
+
+        // Sort wallets by non-added wallets then added wallets
+        modified_wallets.sort((a, b) => {
+            if (a.is_added !== b.is_added) return a.is_added ? 1 : -1;
+
+            return 0;
+        });
+
+        // Sort the added wallets alphabetically by fiat, crypto, then virtual (if any)
+        modified_wallets.sort((a, b) => {
+            if (!a.is_added || !b.is_added) return 0;
+            if (a.is_crypto !== b.is_crypto) return a.is_crypto ? 1 : -1;
+
+            return (a.currency || 'USD').localeCompare(b.currency || 'USD');
```

Comment:
```suggestion
            if (!a.is_added || !b.is_added) return 0;
            if (a.is_crypto !== b.is_crypto) return a.is_crypto ? 1 : -1;

            return (a.currency || 'USD').localeCompare(b.currency || 'USD');
```
This sorting function is similar to the one below. You can split this function to its own declaration and reuse it in both cases.